### PR TITLE
Allow disabling updates across major versions with Github strategy

### PR DESCRIPTION
**Pull request by padraic**
A little cross-pollination from #37 to keep features as aligned as possible within the existing behaviour.

### CODE REVIEW

**Comment by keradus** [comment]
Hi ! looks like a change i could benefit from ;)
What is the state here?

**Comment by theofidry** [comment]
The state is I'm a bit short of time for Christmas and NY so I'm trying to finish to write an article with a roadmap regarding the PHAR-related projects (which include this one) and then I'll tackle them.

Not sure where this one will be in the priority list, I think I'll want to finish PHP-Scoper integration with box first.

Also I find the current usage of this library a bit cumbersome for simple cases so I would like to change a couple of things there as well (see https://github.com/humbug/phar-updater/issues/53).

TL:DR; either wait for my post (in 1-2 weeks) or a couple of month, I can make sure to ping you for this when it's done :)

**Comment by keradus** [comment]
Don't worry, I'm not rushing you ;)
Just asking about **current** state of art here, as this PR was approved half year ago ;) (also, it is not issue of christmas time for that ;P)

thanks for feedback !
